metrics/prometheus.go

```diff
@@ -2060,3 +2118,23 @@ func getMinCoreScalingRatio(s *info.ContainerStats) metricValues {
 	}
 	return values
 }
+
+func getPSIValues(s *info.ContainerStats, psi *info.PSIStats, psiMetric string) metricValues {
+	v := make(metricValues, 0, 2)
+	switch psiMetric {
+	case "avg10":
+		v = append(v, metricValue{value: psi.Some.Avg10, timestamp: s.Timestamp, labels: []string{"some"}})
+		v = append(v, metricValue{value: psi.Full.Avg10, timestamp: s.Timestamp, labels: []string{"full"}})
+	case "avg60":
+		v = append(v, metricValue{value: psi.Some.Avg60, timestamp: s.Timestamp, labels: []string{"some"}})
+		v = append(v, metricValue{value: psi.Full.Avg60, timestamp: s.Timestamp, labels: []string{"full"}})
+	case "avg300":
+		v = append(v, metricValue{value: psi.Some.Avg300, timestamp: s.Timestamp, labels: []string{"some"}})
+		v = append(v, metricValue{value: psi.Full.Avg300, timestamp: s.Timestamp, labels: []string{"full"}})
+	case "total":
+		// total is measured as microseconds
+		v = append(v, metricValue{value: float64(time.Duration(psi.Some.Total)*time.Microsecond) / float64(time.Second), timestamp: s.Timestamp, labels: []string{"some"}})
+		v = append(v, metricValue{value: float64(time.Duration(psi.Full.Total)*time.Microsecond) / float64(time.Second), timestamp: s.Timestamp, labels: []string{"full"}})
```

Comment:
Note for CPU, we don't need to expose "full". In practice, I've only found the "some" metrics to be useful. The "some" value is a superset of "full". IMO we should just include that to reduce the cardinality.

See the [PSI docs](https://docs.kernel.org/accounting/psi.html):

> CPU full is undefined at the system level, but has been reported since 5.13, so it is set to zero for backward compatibility.

Reply:
I would like to raise my opinion for exposing "CPU full" metrics on a container level.

While "CPU full" is undefined at the system level, it reports values for other cgroups:
```console
$ cat /sys/fs/cgroup/user.slice/cpu.pressure
some avg10=0.00 avg60=0.05 avg300=0.20 total=68139509
full avg10=0.00 avg60=0.00 avg300=0.00 total=40148380
```

Exposing both might be useful to differentiate between cgroups that try to execute more work then compute time available to them (indicated through "CPU some") and cgroups which are fully blocked/stalled (maybe because of other cgroups and/or process priority, …; indicated through "CPU full").

Reply:
Interesting, thanks for this info. Are there kernel docs that document the per-cgroup behavior?

IMO this should be a separate metric name, rather than a label. The reason is that since `some` is a inclusive of `full`, doing something like `sum(rate(container_pressure_cpu_seconds_total[1m]))` would be confusing.

I would suggest these two metric names:
* `container_pressure_cpu_seconds_total`
* `container_pressure_cpu_full_seconds_total`